bin/stt_test/stt_providers/vosk/Vosk.py

```diff
@@ -0,0 +1,58 @@
+
+import logging
+from pathlib import Path
+import speech_recognition as sr  # type: ignore
+import ast
+from stt_test.utils.files import download_and_extract, prepare_dir
+from stt_test.stt_providers.ISTTBase import ISSTBase
+import os
+import shutil
+from vosk import Model, KaldiRecognizer  # type: ignore
+
+
+class Vosk(ISSTBase):
+
+    MODEL_SIZES = ['small', 'medium', 'big', 'large']
+    MODEL_PATH = Path(os.path.join(os.path.dirname(
+        os.path.realpath(__file__)), 'model'))
+    _model: KaldiRecognizer
+
+    def _download_model_and_extract(self, model_size: str):
+        match model_size:
+            case 'small':
+                url = 'https://alphacephei.com/vosk/models/vosk-model-small-en-us-0.15.zip'
```

Comment:
Seems like base url can be extracted to a variable

Reply:
Yes, it can, but I think that it will only increase the amount of code, operations and reduce readability

Reply:
`model_url = 'https://alphacephei.com/vosk/models/vosk-model-small-en-us-0.15.zip'
'model_url2 = 'https://alphacephei.com/vosk/models/vosk-model-small-en-us-0.15.zip'
`
`base_url='https://alphacephei.com/vosk/models'
model_url = base_url + 'vosk-model-small-en-us-0.15.zip'
'model_url2 = base_url + 'vosk-model-small-en-us-0.15.zip'
`
In my honest opinion, the second one is way more readable even if there is one code line more. It's easier to catch what is the exact name of the model because the rest of the link is hidden in the variable - less data to process by eyes. What do you think?

Reply:
Maybe a look-up table name -> url is a better choice there instead of match?